File: 兴趣班入门百练讲义/NQ100/NQ035/NQ035_improved.cpp
// NQ035 二进制字符串奇偶位互换
// 将二进制字符串的相邻两位（奇偶位）互换位置
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    int test_case_count;
    cin >> test_case_count;  // 读取测试用例数量
    
    while (test_case_count--) {  // 遍历每个测试用例
        string binary_str;
        cin >> binary_str;  // 读取二进制字符串
        
        // 每两位互换位置（奇偶位互换）
        // 遍历步长为2，每次处理i和i+1位置的字符
        for (int i = 0; i < binary_str.length() - 1; i += 2) {
            swap(binary_str[i], binary_str[i + 1]);
        }
        
        cout << binary_str << endl;  // 输出互换后的二进制字符串
    }
    
    return 0;
}

File: source/algorithm/NQ035/NQ035_improved.py
# NQ035 二进制字符串奇偶位互换
# 将二进制字符串的相邻两位（奇偶位）互换位置
def main():
    test_case_count = int(input())  # 读取测试用例数量
    
    for _ in range(test_case_count):  # 遍历每个测试用例
        binary_str = input().strip()  # 读取二进制字符串
        chars = list(binary_str)  # 转换为字符列表以便修改
        
        # 每两位互换位置（奇偶位互换）
        # 遍历步长为2，每次处理i和i+1位置的字符
        for i in range(0, len(chars) - 1, 2):
            chars[i], chars[i + 1] = chars[i + 1], chars[i]
        
        print(''.join(chars))  # 转换回字符串并输出

if __name__ == "__main__":
    main()